Core/Core/Use Cases/UploadManager.swift

```diff
@@ -379,6 +379,7 @@ open class UploadManager: NSObject, URLSessionDelegate, URLSessionTaskDelegate,
     public func urlSessionDidFinishEvents(forBackgroundURLSession session: URLSession) {
         completionHandler?()
```

Comment:
should this happen after invalidating the session. On the off change it kicks off more stuff.